benchmark/dataset.py

```diff
@@ -41,9 +41,11 @@ def download(self):
             print(f"Downloading {self.config.link}...")
             tmp_path, _ = urllib.request.urlretrieve(self.config.link)
 
-            if tmp_path.endswith(".tgz") or tmp_path.endswith(".tar.gz"):
+            if self.config.link.endswith(".tgz") or self.config.link.endswith(
```

Comment:
I might be picky but I don't see how it is better than the previous row.

Reply:
I'm not sure about the other platforms, but on Linux urlretrieve puts the data into a file with a random name, without even keeping the original extension. The previous version was checking the extension of the randomly generated name, so the condition was never true.

Reply:
ouch, I occidentally publish these 2 comments, they were just drafts :(